gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp

```diff
@@ -396,17 +419,19 @@ void GazeboRosControlPlugin::Load(gazebo::physics::ModelPtr parent, sdf::Element
               controller_type);
             impl_->controllers_.push_back(controller);
             load_params_from_yaml(
-              controller->get_lifecycle_node(),
+              controller->get_node(),
               impl_->param_file_,
               controller_name);
-            if (controller_name != "joint_state_controller") {
-              controller->get_lifecycle_node()->configure();
+            if (controller_name == "joint_state_controller") {
+              controller->assign_interfaces({}, std::move(loaned_state_ifs));
             }
```

Comment:
this seems very error prone. Why does the gazebo plugin have to deal with controllers? All that logic should be hidden within the controller manager.

Reply:
happy to remove some code